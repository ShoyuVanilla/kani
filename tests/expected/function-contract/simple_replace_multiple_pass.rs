// Copyright Kani Contributors
// SPDX-License-Identifier: Apache-2.0 OR MIT
// kani-flags: -Zfunction-contracts

#[kani::requires(divisor != 0)]
#[kani::ensures(|result : &u32| *result <= dividend)]
fn div(dividend: u32, divisor: u32) -> u32 {
    dividend / divisor
}

#[kani::requires(a >= b)]
#[kani::ensures(|result : &u32| *result <= a)]
fn sub(a: u32, b: u32) -> u32 {
    a - b
}

#[kani::proof]
#[kani::stub_verified(div)]
#[kani::stub_verified(sub)]
fn main() {
    assert!(div(sub(9, 1), 1) != 10, "contract guarantees smallness");
}
