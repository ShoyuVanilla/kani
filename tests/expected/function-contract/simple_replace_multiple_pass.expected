.assertion\
- Status: SUCCESS\
- Description: "divisor != 0"

.assertion\
- Status: SUCCESS\
- Description: "a >= b"

main.assertion\
- Status: SUCCESS\
- Description: ""contract guarantees smallness""

VERIFICATION:- SUCCESSFUL
